acceptor/tcp_acceptor.go

```diff
@@ -78,21 +77,26 @@ func (t *tcpPlayerConn) GetNextMessage() (b []byte, err error) {
 
 // NewTCPAcceptor creates a new instance of tcp acceptor
 func NewTCPAcceptor(addr string, certs ...string) *TCPAcceptor {
-	keyFile := ""
-	certFile := ""
+	certificates := []tls.Certificate{}
 	if len(certs) != 2 && len(certs) != 0 {
 		panic(constants.ErrInvalidCertificates)
 	} else if len(certs) == 2 {
-		certFile = certs[0]
-		keyFile = certs[1]
+		cert, err := tls.LoadX509KeyPair(certs[0], certs[1])
+		if err != nil {
+			panic(constants.ErrInvalidCertificates)
+		}
+		certificates = append(certificates, cert)
 	}
 
+	return NewTCP(addr, certificates...)
+}
+
+func NewTCP(addr string, certs ...tls.Certificate) *TCPAcceptor {
```

Comment:
This name seems fairly generic, could we name it something like NewTLSAcceptor something like that?

Reply:
That's a good question, I always struggle with naming in Golang.

I defined it as `NewTCP` so that it doesn't stutter when we have it on code like `acceptor.NewTCP` (instead of `acceptor.NewTCPAcceptor`). The more verbose option would also conflict with the previous existing definition.

Another option could be `NewTCPWithCertificates`.

What do you think?

Reply:
Given the current naming standards, I agree with Rafa on changing to NewTLSAcceptor